File: scr/Catalog.java
public enum Catalog {
    findAllOdds(0, Solution000.class, Status.accepted),
    twoSum(1, Solution001.class, Status.accepted),
    addTwoNumbers(2, Solution002.class, Status.accepted),
    longestSubstringWithoutRepeatingCharacters(3, Solution003.class, Status.accepted),
    longestPalindromicSubstring(5, Solution005.class, Status.accepted),
    zigZagConversion(6, Solution006.class, Status.accepted),
    reverseInteger(7, Solution007.class, Status.accepted),
    palindromeNumber(9, Solution009.class, Status.accepted),
    regularExpressionMatching(10, Solution010.class, Status.accepted),
    containerWithMostWater(11, Solution011.class, Status.accepted),
    longestCommonPrefix(14, Solution014.class, Status.accepted),
    threeSum(15, Solution015.class, Status.timeLimitExceeded),
    letterCombinationsOfAPhoneNumber(17, Solution017.class, Status.accepted),
    removeNthNodeFromEndOfList(19, Solution019.class, Status.accepted),
    validParentheses(20, Solution020.class, Status.accepted),
    mergeTwoSortedLists(21, Solution021.class, Status.accepted),
    generateParentheses(22, Solution022.class, Status.accepted),
    mergeKSortedLists(23, Solution023.class, Status.accepted),
    swapNodesInPairs(24, Solution024.class, Status.accepted),
    reverseNodesInKGroup(25, Solution025.class, Status.accepted),
    removeDuplicatesFromSortedArray(26, Solution026.class, Status.accepted),
    removeElement(27, Solution027.class, Status.accepted),
    implementStrStr(28, Solution028.class, Status.accepted),
    divideTwoIntegers(29, Solution029.class, Status.accepted),
    nextPermutation(31, Solution031.class, Status.accepted),
    searchInRotatedSortedArray(33, Solution033.class, Status.accepted),
    searchForARange(34, Solution034.class, Status.accepted),
    searchInsertPosition(35, Solution035.class, Status.accepted),
    validSudoku(36, Solution036.class, Status.accepted),
    sudokuSolver(37, Solution037.class, Status.accepted);

    int number;
    Class solution;
    String status;

    Catalog(int number, Class solution, Status status) {
        this.number = number;
        this.solution = solution;
        this.status = status.s;
    }

}

File: Catalog.kt
val Java = "Java"
val Kotlin = "Kotlin"
val Scala = "Scala"


enum class S {
    Accepted,
    TimeLimitExceeded,
    WrongAnswer,
    ;
}


enum class Catalog(val language: String, val problem: Int, val solution: S) {
    FindAllOdds(Kotlin, 0, S.Accepted),
    TwoSum(Kotlin, 1, S.Accepted),
    AddTwoNumbers(Kotlin, 2, S.Accepted),
    LongestSubstringWithoutRepeatingCharacters(Kotlin, 3, S.Accepted),
    LongestPalindromicSubstring(Kotlin, 5, S.Accepted),
    ZigZagConversion(Kotlin, 6, S.Accepted),
    ReverseInteger(Kotlin, 7, S.Accepted),
    PalindromeNumber(Kotlin, 9, S.Accepted),
    RegularExpressionMatching(Kotlin, 10, S.Accepted),
    ContainerWithMostWater(Kotlin, 11, S.Accepted),
    LongestCommonPrefix(Kotlin, 14, S.Accepted),
    ThreeSum(Kotlin, 15, S.Accepted),
    LetterCombinationsOfAPhoneNumber(Kotlin, 17, S.Accepted),
    RemoveNthNodeFromEndOfList(Kotlin, 19, S.Accepted),
    ValidParentheses(Kotlin, 20, S.Accepted),
    MergeTwoSortedLists(Kotlin, 21, S.Accepted),
    GenerateParentheses(Kotlin, 22, S.Accepted),
    MergeKSortedLists(Kotlin, 23, S.Accepted),
    SwapNodesInPairs(Kotlin, 24, S.Accepted),
    ReverseNodesInKGroup(Kotlin, 25, S.Accepted),
    RemoveDuplicatesFromSortedArray(Kotlin, 26, S.Accepted),
    RemoveElement(Kotlin, 27, S.Accepted),
    ImplementStrStr(Kotlin, 28, S.Accepted),
    DivideTwoIntegers(Kotlin, 29, S.Accepted),
    NextPermutation(Kotlin, 31, S.Accepted),
    LongestValidParentheses(Kotlin, 32, S.Accepted),
    SearchInRotatedSortedArray(Kotlin, 33, S.Accepted),
    SearchForARange(Kotlin, 34, S.Accepted),
    SearchInsertPosition(Kotlin, 35, S.Accepted),
    ValidSudoku(Kotlin, 36, S.Accepted),
    SudokuSolver(Kotlin, 37, S.Accepted),
    CombinationSum(Kotlin, 39, S.Accepted),
    CombinationSum2(Kotlin, 40, S.Accepted),
    FirstMissingPositive(Kotlin, 41, S.Accepted),
    TrappingRainWater(Kotlin, 42, S.Accepted),
    WildcardMatching(Kotlin, 44, S.TimeLimitExceeded),
    JumpGame2(Kotlin, 45, S.Accepted),
    Permutations(Kotlin, 46, S.Accepted),
    Permutations2(Kotlin, 47, S.Accepted),
    RotateImage(Kotlin, 48, S.Accepted),

    JumpGame(Scala, 55, S.Accepted),
    MergeIntervals(Scala, 56, S.Accepted),
    InsertInterval(Scala, 57, S.Accepted),
    ;

}
